pipeline_steps/release.groovy

```diff
@@ -0,0 +1,49 @@
+void configure_git(){
+  sh """/bin/bash -xe
+    git config --global user.email "rpc-jenkins-svc@github.com"
+    git config --global user.name "rpc.jenkins.cit.rackspace.net"
+  """
+}
+
+void clone(String repo_url){
+  print "Cloning ${repo_url} to ./repo"
+  // using sh to clone so the ssh agent creds can be used.
+  sshagent (credentials:['rpc-jenkins-svc-github-ssh-key']){
+    sh """/bin/bash -xe
+      mkdir -p ~/.ssh
+      ssh-keyscan github.com >> ~/.ssh/known_hosts
+      git clone "${repo_url}" repo
+    """
+  }
+}
+
+void tag(String version, String ref){
+  print "Tagging ${version} at ${ref}"
+  sshagent (credentials:['rpc-jenkins-svc-github-ssh-key']){
+    sh """/bin/bash -xe
+      cd repo
+      mkdir -p ~/.ssh
+      ssh-keyscan github.com >> ~/.ssh/known_hosts
+      git fetch --all --tags
+      git tag -a -m \"$version\" \"$version\" origin/$ref
```

Comment:
Can never quite remember -- but does this need to be `${version}` and `${ref}`?

Reply:
@mattt416 Are you asking due to this being a bash script in a groovy script? I think if we use {} here we may end up triggering a groovy macro?

Reply:
@odyssey4me Yep, that's what I'm referring to ... the function below (`update_rc_branch`) uses `${rc}` and `${mainline}`, for example.

Reply:
Yeah, itd be better if it was consistent. 👍

Reply:
Done